.github/workflows/release.yml

```diff
@@ -0,0 +1,104 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'kms-v*'
+
+permissions:
+  contents: write
+  actions: write
+  checks: write
+  packages: write
+  id-token: write
+
+jobs:
+  tests:
+    name: Tests
+    uses: ./.github/workflows/ci.yml
+    secrets:
+      MANAGED_ID_CLIENT_ID: ${{ secrets.MANAGED_ID_CLIENT_ID }}
+      MANAGED_ID_TENANT_ID: ${{ secrets.MANAGED_ID_TENANT_ID }}
+
+  release:
+    needs: tests
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set release version
+      id: vars
+      run: echo "release_version=${GITHUB_REF#refs/tags/kms-v}" >> $GITHUB_ENV
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: '22'
+
+    - name: Install dependencies
+      run: npm install
+
+    - name: Build project
+      run: make build
+
+    - name: Generate SBOM
+      run: npx cyclonedx-bom -o sbom.xml
+
+    - name: Zip /dist directory
+      run: zip -r dist.zip dist
+
+    - name: Fetch PR notes
+      id: pr_notes
+      run: |
+        echo "## Release Notes" > RELEASE_NOTES.md
+        echo "" >> RELEASE_NOTES.md
+        PRS=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
```

Comment:
Another way we could do this is enforce squash and merge on PRs and then inspect the commit history

This will catch changes done directly to main

Reply:
done